.github/workflows/build.yml

```diff
@@ -157,18 +165,20 @@ jobs:
       cmake_preset_type: ${{needs.common_config.outputs.cmake_preset_type_resolved}}
       cibw_image_tag: ${{needs.cibw_docker_image.outputs.tag}}
       matrix: ${{needs.common_config.outputs.linux_matrix}}
+      distro: ${{needs.start_linux_ec2_runner.outputs.label}}
 
   leader-cpp-test-linux:
-    # Compile and run the C++ tests separately concurrently with the following job
     needs: [leader-compile-linux, cibw_docker_image, common_config]
     name: Linux C++ Tests
     uses: ./.github/workflows/build_steps.yml
     secrets: inherit
+    permissions: {packages: write}
     with:
       job_type: cpp-tests
       cmake_preset_type: ${{needs.common_config.outputs.cmake_preset_type_resolved}}
       cibw_image_tag: ${{needs.cibw_docker_image.outputs.tag}}
       matrix: ${{needs.common_config.outputs.linux_matrix}}
+      distro: ubuntu-latest
```

Comment:
Please also remove the now redundant `distro` fields from `common_config` and [here](https://github.com/man-group/ArcticDB/blob/64d5e738b275dfa243527f5723c62861daa39053/.github/workflows/build_steps.yml#L25) to avoid confusion.